src/components/TextArea/TextArea.test.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { render, screen, fireEvent, cleanup } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import TextArea from './TextArea';
+
+afterEach(cleanup);
+
+describe('TextArea', () => {
+  test('renders the correct label text', () => {
+    render(<TextArea labelText="Label text" placeholder="Input text" />);
+
+    expect(screen.getByText('Label text'));
+  });
+  test('renders a textarea input', () => {
+    render(<TextArea labelText="Label text" placeholder="Input text" />);
+
+    expect(TextArea).toBeDefined();
+  });
+  test('renders the correct placeholder text', () => {
+    render(<TextArea labelText="Label text" placeholder="Input text" />);
+
+    expect(screen.getByPlaceholderText('Input text'));
+  });
+  test('calls onChange prop when text changed', () => {
+    const handleChange = jest.fn();
+    render(
+      <TextArea
+        id="textarea"
+        labelText="label text"
+        placeholder="input text"
+        value="original text"
+        onChange={handleChange}
+      />
+    );
+    screen.debug();
+    fireEvent.change(screen.getByRole('textbox'), {
+      target: { value: 'modified text' }
+    });
+    screen.debug();
+    expect(handleChange).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
Not sure I need all these tests at this stage? I'm not sure If the last test is right. I was trying to simulate modifying the text so that handleChange fires. Screen.debug() doesn't show that the text has been changed.

Reply:
You can get rid of all the ones that are testing the rendering, the onChange function is the important thing to test. You can clean this up significantly by creating a stateful mock component like so:
```
import React, { useState } from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TextArea from './TextArea';

const MockInput = () => {
  const [value, setValue] = useState('');

  const handleChange = (event) => {
    setValue(event.target.value);
  };

  return (
    <TextArea
      name="test"
      id="test"
      value={value}
      labelText="Label"
      onChange={handleChange}
    />
  );
};

const setup = () => {
  const utils = render(<MockInput />);
  const input = utils.getByLabelText('Label');
  return {
    input,
    ...utils
  };
};

test('It should change value when new input received', () => {
  const { input } = setup();
  fireEvent.change(input, { target: { value: 'Hey Cath!' } });
  expect(input.value).toBe('Hey Cath!');
});
```

The example of this is [here](https://testing-library.com/docs/example-input-event/).

@bristol-tech-fair/webdev-101 I'm tagging you all so you can see how to test change handlers.

The key thing to remember is that test files are just JS, so you put whatever you want in there. I've basically built a fake component with state and am then using that to test the onChange functionality.